lib/scraped_page_archive.rb

```diff
@@ -15,6 +15,7 @@ class ScrapedPageArchive
   class Error < StandardError; end
 
   attr_writer :github_repo_url
+  attr_writer :git
```

Comment:
Adding this to the public interface rings all sorts of alarm bells for me. This is essentially saying that we don't want to be totally coupled to the `Git` implementation we're using, and people can swap in their own. But we're providing that at an odd point, as there's lots of other logic bound up in the `git` method later on. If we were to document what we want people to pass in here, it seems like it would be quite hard to do. 

I don't know what the correct approach here is, but if we need to loosen the coupling to a hard-coded `Git` below, this doesn't quite seem like the right place or right way to do it.